metadata-ingestion/src/datahub/ingestion/source/kafka.py

```diff
@@ -202,7 +202,15 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
         for t, t_detail in topics.items():
             self.report.report_topic_scanned(t)
             if self.source_config.topic_patterns.allowed(t):
-                yield from self._extract_record(t, t_detail, extra_topic_details.get(t))
+                try:
+                    yield from self._extract_record(
+                        t, t_detail, extra_topic_details.get(t)
+                    )
+                except Exception as e:
+                    logger.warning(f"Failed to extract topic {t}", exc_info=True)
+                    self.report.report_warning(
```

Comment:
should this be a failure?

Reply:
Idk about this source at all, so I erred on the side of caution lol. I am happy to swap if you think it should be